From 7567faf4adaea6298cf5244176a88506e5e26bc7 Mon Sep 17 00:00:00 2001
From: Matt Wittmann <mwittmann@relaytx.com>
Date: Mon, 6 Mar 2023 08:09:00 -0800
Subject: [PATCH 2/3] Adapt cmake build

* Use deps from nix

* Parameterize python version

* Remove CUDA_ARCH setting

* Remove -Werror (warnings from eigen)
---
 timemachine/cpp/CMakeLists.txt | 29 +++++------------------------
 1 file changed, 5 insertions(+), 24 deletions(-)

diff --git a/timemachine/cpp/CMakeLists.txt b/timemachine/cpp/CMakeLists.txt
index 97114ce0..1192fc5b 100644
--- a/timemachine/cpp/CMakeLists.txt
+++ b/timemachine/cpp/CMakeLists.txt
@@ -2,11 +2,13 @@ cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(PYBIND11_PYTHON_VERSION 3.10 CACHE STRING "Which version of python we're building wrappers against")
+set(PYBIND11_PYTHON_VERSION @pythonVersion@ CACHE STRING "Which version of python we're building wrappers against")
 
 project(timemachine LANGUAGES CXX CUDA)
 
-find_package(Python 3.10 COMPONENTS Interpreter Development REQUIRED)
+find_package(Python @pythonVersion@ COMPONENTS Interpreter Development REQUIRED)
+find_package(Eigen3 3.4 REQUIRED NO_MODULE)
+find_package(pybind11)
 
 set(DEBUG_BUILD "$ENV{DEBUG_BUILD}")
 
@@ -18,7 +20,6 @@ else()
 endif()
 
 
-string(APPEND CMAKE_CXX_FLAGS "${COMMON_COMPILER_FLAGS} -Werror")
 string(APPEND CMAKE_CUDA_FLAGS "${COMMON_COMPILER_FLAGS}")
 
 string(APPEND CMAKE_CUDA_FLAGS
@@ -40,19 +41,6 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif()
 
 set(LIBRARY_NAME custom_ops)
-set(PYBIND_SRC_DIR pybind11)
-
-if(NOT EXISTS ${PYBIND_SRC_DIR})
-  execute_process(COMMAND git clone --depth 1 --branch v2.9.2 https://github.com/pybind/pybind11.git ${PYBIND_SRC_DIR})
-endif()
-
-set(EIGEN_SRC_DIR eigen)
-if(NOT EXISTS ${EIGEN_SRC_DIR})
-  execute_process(COMMAND git clone --depth 1 --branch 3.3.9 https://gitlab.com/libeigen/eigen.git ${EIGEN_SRC_DIR})
-endif()
-
-add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR} ${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR})
-
 
 # Disable warnings on vendored files
 file(
@@ -127,17 +115,10 @@ pybind11_add_module(${LIBRARY_NAME} SHARED NO_EXTRAS
 
 include_directories(src/kernels)
 include_directories(SYSTEM ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
-include_directories(SYSTEM ${CMAKE_CURRENT_BINARY_DIR}/${EIGEN_SRC_DIR})
+include_directories(${EIGEN3_INCLUDE_DIRS})
 
 set_property(TARGET ${LIBRARY_NAME} PROPERTY CUDA_STANDARD 17)
 
-if (NOT CUDA_ARCH)
-  set(CUDA_ARCH "native")
-endif()
-message("Cuda Arch: " ${CUDA_ARCH})
-set_property(TARGET ${LIBRARY_NAME} PROPERTY CUDA_ARCHITECTURES "${CUDA_ARCH}")
-unset(CUDA_ARCH CACHE)
-
 target_link_libraries(${LIBRARY_NAME} PRIVATE -lcurand)
 set_target_properties(${LIBRARY_NAME} PROPERTIES PREFIX "")
 
-- 
2.47.0

